steps/user_permissions.py

```
@@ -79,7 +81,28 @@ def user_logs_into_app(context, user_name, app_view):
         context.helpers.config.get('urls').get('{}_login'.format(app_view))
     login_page.driver.get(login_url)
     username = get_user_credentials(context.client, user_name)
+    is_desktop = app_view == 'desktop'
     if username:
-        login_page.login(username, username)
+        login_page.login(username, username, desktop=is_desktop)
     else:
         raise ValueError('No user found')
+    login_page.verify_page_loaded(app_view)
+
+
+@then("the available menu items are filtered for the {user_role} role")
+def verify_correct_desktop_menu_items_available(context, user_role):
+    """
+
+    :param context:
+    :param user_role:
+    :return:
+    """
+    pages_for_role = []
+    for row in context.table:
+        pages_for_role.append(row.get('page'))
+    page_list = BaseDesktopPage(context.driver)
+    pages = page_list.get_menu_items_list()
+    for page in pages:
+        page_name = page_list.get_menu_item_text(page)
+        assert (page_name in pages_for_role), \
+            "'{}' page not in {}".format(page_name, pages_for_role)
```

I'm don't think the brackets are necessary here but I could be wrong.

---

Also I'm a big fan of using the terms _expected_ and _actual_ when doing assertions. In my opinion it would make it slightly clearer if it read `assert actual_page_name in expected_pages_for_role`.

---

Also this will fail on the first menu item it encounters that is not as expected. That means that it will hide other failures if there are more. It might be that we're fine with that in E2E tests, its for us to decide. But an alternative is to build the entire list of _actual_ and then assert it matches the entire list of _expected_.

---

After discussion with @moderatemisbehaviour , tech debt EOBS-2337 has been raised for this, as it'll need a new library to be added to use that style of assertions.